### PR TITLE
feat: AliPay frontend implementation(API Keys and Buttons)

### LINKED ISSUE #3023: Integration of AliPay Frontend
**Describe the Feature**
This serves as a parent issue alluding to the frontend implementation of the AliPay Payment Gateway.
This issue includes the settings inclusion of AliPay Gateway, basic-wizard-setup inclusion and linkage of the frontend to the server.

**Additional context**
Working on this.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
alipay-final

Part of #3023
Server PR https://github.com/fossasia/open-event-server/pull/5960

#### Short description of what this resolves:
- This feature includes AliPay options in the settings, along with API/secret key fields for admins
-Also includes the AliPay option while placing orders

#### Changes proposed in this pull request:

- Settings options - AliPay
- Publishable/secret keys 
- Additional Model Fields

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 1]
```diff
@@ -309,6 +309,25 @@
...
+              {{t 'Yes, accept payment through AliPay Gateway'}}
+              <div class="ui hidden divider"></div>
+              <span class="text muted">
+                {{t 'AliPay can accept Credit and Debit Cards , Net-Banking and AliPay. Find more details '}}
+                <a href="https://stripe.com/docs/sources/alipay" target="_blank" rel="noopener noreferrer">{{t 'here'}}</a>.
+              </span>
+            </label>
+          </div>
+        </div>
+        {{#if data.event.canPayByOmise}}
```
Comment (kushthedude) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: @mrsaicharan1 Why the following `Comparator` is being used in here ?

File: app/components/forms/wizard/basic-details-step.js [hunk 2]
```diff
@@ -48,6 +48,10 @@ export default Component.extend(FormMixin, EventWizardMixin, {
     return this.get('settings.isOmiseActivated') && find(paymentCurrencies, ['code', this.get('data.event.paymentCurrency')]).omise;
   }),
 
+  canAcceptAliPay: computed('data.event.paymentCurrency', 'settings.isAliPayActivated', function() {
```
Comment (kushthedude) [inline app/components/forms/wizard/basic-details-step.js]: @mrsaicharan1 Please add the corresponding field for `AliPay` in the Dictionary file too .

File: app/templates/components/forms/wizard/basic-details-step.hbs (hunk 1 above)
Comment (kushthedude) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: Why the following `Comparator` is being used in here ?

File: app/components/forms/wizard/basic-details-step.js (hunk 2 above)
Comment (mrsaicharan1) [reply app/components/forms/wizard/basic-details-step.js]: Dictionary? Where do you mean?

Comment (mrsaicharan1) [reply app/components/forms/wizard/basic-details-step.js]: Thanks for pointing that out @kushthedude

File: app/utils/dictionary/payment.js [hunk 3]
```diff
@@ -31,6 +34,7 @@ export const paymentCurrencies = [
     symbol : 'R$',
     name   : 'Brazilian real',
     stripe : true,
+    alipay : true,
```
Comment (kushthedude) [inline app/utils/dictionary/payment.js]: @mrsaicharan1 AliPay doesn't support all the currencies you have added `true` to , Please take this as reference and make following changes
Screenshot 2019-05-29 at 4 59 08 AM

Comment (mrsaicharan1) [reply app/utils/dictionary/payment.js]: @kushthedude Yeah, I just checked it out. It was using Stripe Sources , but I checked it out later.

Comment (kushthedude) [reply app/utils/dictionary/payment.js]: Cool , Make appropriate changes 😄

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 @shreyanshdwivedi @uds5501 Please review the flow for the frontend. Once this is merged, will open another PR which will send the `order_id` to  the backend using the `loader` service. Don't want to cramp everything in one PR. 
Here's the related server PR: https://github.com/fossasia/open-event-server

**Comment by kushthedude** [comment]
> This looks good. One minor doubt, the link to get the Alipay keys is redirecting to stripe, is it correct?

@uds5501 Yeah , Thats Correct , We are connecting` AliPay` using `Stripe Resources`

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 Please have a look

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi Would be great if you could take over after this is merged. 
After this, the frontend still needs to be linked to the server route given in my server PR.

**Comment by shreyanshdwivedi** [comment]
@mrsaicharan1 sure. I'll take it forward in a follow-up PR

**Comment by mrsaicharan1** [comment]
```
@alipay_blueprint.route('/create_source/<string:order_identifier>', methods=['GET', 'POST'])
def create_source(order_identifier):
    order = safe_query(db, Order, 'identifier', order_identifier, 'identifier')
    source_object = AliPayPaymentsManager.create_source(amount=order.amount, currency='usd', redirect_return_uri=url_for('alipay_blueprint.alipay_return_uri',  _external=True))
    session['source_id'] = source_object.id
    return redirect(source_object.redirect['url'])
```

```

  actions: {
    aliPayCheckout(order_identifier) {
     this.loader.load(`alipay/create_source/${order_identifier}`)
        .then(link => {
          window.location.replace(link);
        });

    }
```

I'm trying to redirect from the `loader` service in the frontend, but due to CSP, I'm not able to. Alipay requires redirection. Can you suggest something for this @CosmicCoder96 @uds5501 @shreyanshdwivedi ?

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 Please review.

**Comment by mariobehling** [comment]
Please provide visuals/screenshots/images - for the changes right on the top!

**Comment by mrsaicharan1** [comment]
@mariobehling I've updated with the GIF of the implementation. Please check the video.

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 @niranjan94 Please review.

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 @uds5501 @shreyanshdwivedi  Resolved conflicts. Please review again

File: app/components/forms/admin/settings/payment-gateway-form.js [hunk 4]
```diff
@@ -134,6 +153,10 @@ export default Component.extend(FormMixin, {
     return this.get('settings.omiseTestPublic') || this.get('settings.omiseLivePublic');
   }),
 
+  isCheckedAliPay: computed(function() {
+    return this.get('settings.aliPaySecretKey') || this.get('settings.aliPayPublishableKey');
```
Comment (CosmicCoder96) [inline app/components/forms/admin/settings/payment-gateway-form.js]: will this implementation work if either of these keys are missing?

File: app/components/forms/admin/settings/payment-gateway-form.js [hunk 5]
```diff
@@ -152,6 +175,12 @@ export default Component.extend(FormMixin, {
             'paypalClient'        : null
           });
         }
+        if (this.isCheckedAliPay === false) {
```
Comment (CosmicCoder96) [inline app/components/forms/admin/settings/payment-gateway-form.js]: no point in comparing something to false explicitly, use !.

File: app/components/forms/admin/settings/payment-gateway-form.js [hunk 6]
```diff
@@ -134,6 +153,10 @@ export default Component.extend(FormMixin, {
     return this.get('settings.omiseTestPublic') || this.get('settings.omiseLivePublic');
   }),
 
+  isCheckedAliPay: computed(function() {
```
Comment (niranjan94) [inline app/components/forms/admin/settings/payment-gateway-form.js]: Ensure your computed property _watches_ the keys it depends on ...

File: app/components/forms/admin/settings/payment-gateway-form.js (hunk 5 above)
Comment (mrsaicharan1) [reply app/components/forms/admin/settings/payment-gateway-form.js]: @CosmicCoder96 Done.

File: app/components/forms/admin/settings/payment-gateway-form.js (hunk 6 above)
Comment (mrsaicharan1) [reply app/components/forms/admin/settings/payment-gateway-form.js]: @niranjan94 Passed the keys into the computed property so that it is updated properly

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 @niranjan94 Made the changes. Please review.

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 @niranjan94 The API Keys were getting stored before the latest conflicts & review changes. I don't know what has changed but I'm not able to store them again. :/

**Comment by mrsaicharan1** [comment]
> @CosmicCoder96 @niranjan94 The API Keys were getting stored before the latest conflicts & review changes. I don't know what has changed but I'm not able to store them again. :/
> Here is the relevant server PR [fossasia/open-event-server#5960](https://github.com/fossasia/open-event-server/pull/5960)

Alright, so the problem was I was using `aliPayPublishableKey` where I was supposed to use `alipayPublishableKey`. Field name error. So yeah, It's working fine now :D

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 Please merge

**Pull request merged**